### PR TITLE
Moved cluster-local docs to Services section and renamed 'cluster-local' to 'private'

### LINKED ISSUE #978: Can't find docs on how to create cluster internal-only Knative service
I searched for:

- internal only knative app
- knative services without external endpoint
- knative do not expose

but didn't land anywhere in the docs repo explaining how this can be done. Are we lacking docs on this topic?

/kind bug
/area serving
/area networking

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #978 

## Proposed Changes

- Moved cluster local services section to Services
- Reviewed and edited section
- Added heading to call out example section better

### CODE REVIEW

**Comment by RichieEscarez** [comment]
I was hoping to keep some of the file history for all the content we are moving around. This [stackoverflow](https://stackoverflow.com/questions/433111/how-to-make-git-mark-a-deleted-and-a-new-file-as-a-file-move/433114#433114) suggests using two commits to make your PR look like a moved file vs a "delete and create new". If you are up for it, do you mind trying that out (or otherwise try these steps in any future PR where files are moved)?

Also, similar to your other PR, we need to track down and fix all the hard coded paths to the files being moved (here in the knative/docs repo and in peer repos).

File: docs/serving/services/creating-private-services.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+title: "Creating a private Knative Service"
+linkTitle: "Creating a private Knative Service"
+weight: 20
+type: "docs"
+aliases: [../serving/cluster-local-route.md]
```
Comment (RichieEscarez) [inline docs/serving/services/creating-private-services.md]: does that syntax work? ive not seen it that way but i also dont know if that works

for reference, i know this syntax works: https://raw.githubusercontent.com/knative/docs/master/docs/serving/configuring-autoscaling.md

**Comment by ahmetb** [comment]
> * Renamed from 'cluster-local' to private Services to aid findability for users

This is good for "findability" but technically not so correct.

Ideally, the concept should still be called "cluster-local Service".
But the article title can be Making Services private

**Comment by abrennan89** [comment]
Made updates based on the comments.
@RichieEscarez I sent you a msg on Slack about the moving vs. delete and create comment.

**Comment by abrennan89** [comment]
Closing this PR until we can figure out a proper user story for this one - I think it's adding confusion in the current state.

**Pull request closed without merging**